### ISSUE TITLE
how to transpose a 3D matrix in keras?

### ISSUE THREAD

**Issue by modkzs** [fchollet/keras#7559]
I need to transpose the 1th and 2th dim of  a 3D matrix, but seems that there is no way to limit matrix dim in keras.backend.transpose(like perm in tf.transpose).

**Comment by jaxony** [comment]
Use `permute_dimensions(x, pattern)` instead.
Documentation is here: https://keras.io/backend/

**Comment by JustAnotherUsername123** [comment]
How to use permute_dimensions(x, pattern) in  Keras functional API?
When I try using it, for example like this:
```
...
x = Conv2D(...)(x)
x = permute_dimensions(x, pattern)
...
```
I get: 

> NoneType' object has no attribute '_inbound_nodes'

**Comment by weizhepei** [comment]
> @jaxony How to use permute_dimensions(x, pattern) in Keras functional API?
> When I try using it, for example like this:
> 
> ```
> ...
> x = Conv2D(...)(x)
> x = permute_dimensions(x, pattern)
> ...
> ```
> I get:
> 
> > NoneType' object has no attribute '_inbound_nodes'

Same error. Have you solved this problem?

**Comment by abntu** [comment]
If you want to shuffle dimension in a layer you should use permute layer:
x= keras.layers.permute((2,1,3))(x)

which shuffle the two first dimensions and it leaves the third dimension unchanged.
for more information see:
https://keras.io/layers/core/#permute
